tests/validation/cattlevalidationtest/core/test_machine.py

```diff
@@ -0,0 +1,277 @@
+    machine = wait_for_host(client, machine)
+    host = machine.hosts()[0]
+    assert host.state == 'active'
+    assert machine.accountId == host.accountId
+
+    # Check that the droplet that is being created in Digital Ocean has the
+    # correct configurations
+
+    droplet = check_host_in_digital_ocean(host.ipAddresses()[0].address)
+
+    assert droplet is not None
+    assert droplet["name"] == machine.name
+    assert droplet["image"]["slug"] == expected_values["image"]
+    assert droplet["size_slug"] == expected_values["size"]
+    assert droplet["region"]["slug"] == expected_values["region"]
+
+    # Remove the machine and make sure that the host
+    # and the machine get removed
+
+    machine = client.wait_success(machine.remove())
+    assert machine.state == 'removed'
+
+    host = client.reload(machine.hosts()[0])
+    assert host.state == 'removed'
+
+    wait_for_host_destroy_in_digital_ocean(host.ipAddresses()[0].address)
+
+
+def wait_for_host(client, machine, timeout=DEFAULT_TIMEOUT):
+    start = time.time()
+    time_elapsed = 0
+    machine = client.reload(machine)
+    while len(machine.hosts()) < 1:
+        time.sleep(1)
+        machine = client.reload(machine)
+        time_elapsed = time.time() - start
+        if time_elapsed > timeout:
+            print str(time_elapsed) + " seconds"
+            raise Exception('Timeout waiting for host to be created.')
+
+    host = machine.hosts()[0]
+    host = wait_for_condition(client,
+                              host,
+                              lambda x: x.state == 'active',
+                              lambda x: 'Host state is ' + x.state
+                              )
+
+    return machine
+
+
+def check_host_in_digital_ocean(ipaddress):
+    url = 'https://api.digitalocean.com/v2/droplets'
+    headers = {'Authorization': "Bearer " + access_key}
+    r = requests.get(url, headers=headers)
+    response = r.json()
+    r.close()
+    droplet_list = response["droplets"]
+    matched_droplet = None
+
+    for droplet in droplet_list:
+        if droplet["networks"]["v4"][0]["ip_address"] == ipaddress:
+            matched_droplet = droplet
+            break
+
+    return matched_droplet
+
+
+def delete_host_in_digital_ocean(name):
+    try:
+        url = 'https://api.digitalocean.com/v2/droplets'
+        headers = {'Authorization': "Bearer " + access_key}
+        r = requests.get(url, headers=headers)
+        response = r.json()
+        r.close()
+        droplet_list = response["droplets"]
+
+        for droplet in droplet_list:
+            if droplet["name"] == name:
+                url = 'https://api.digitalocean.com/v2/droplets/' + \
+                      str(droplet["id"])
+                headers = {'Authorization': "Bearer " + access_key}
+                r = requests.delete(url, headers=headers)
+                r.close()
+    except Exception:
+        print "Error encountered when trying to delete machine - " + name
```

Comment:
python has a pretty good/easy-to-use log library. Should use it instead of print.
A good tutorial: https://docs.python.org/2/howto/logging.html

But boils down to:
```
import logging
...
logger = logging.getLogger(__name__)
...
logger.info()
```